### ISSUE TITLE
Split the "ATDL parsing into Java" from the UI

### ISSUE THREAD

**Issue by philipwhiuk** [atdl4j/atdl4j#34]
The library would be more useful to projects if the ATDL parsing into Java classes was split from the UI code and the project available via Maven repositories:

My proposal would be:

1 . Split the code into Maven submodes:

* atdl-core - the ATDL parsing core
* atdl-ui - the UI abstraction layer
* atdl-swt - just the SWT specific code
* atdl-swing - just the Swing specific code
* atdl-all - the drop-in replacement for the current project.

2.  Move the unmarshalling logic from AbstractAtdl4jCompositePanel#parseFixatdlFile to AtdlParser in atdl-core

3. Publish to Maven Central

I'm happy to prepare some PRs to work toward this if it'd be upstreamed.

**Comment by johnnyshields** [comment]
@philipwhiuk I'm looking for a new maintainer for this repo if you're interested.

I'd be fine with this refactor, however, let's keep it all within this same Github repo (i.e. rather than making 5 different Github repos.)

**Comment by philipwhiuk** [comment]
Yep, no need to have multiple repositories. And yes I think I should have time to help maintain it.

**Comment by johnnyshields** [comment]
Ok, great! Please raise the PR first.

**Comment by johnnyshields** [comment]
Also, please use the following package names if possible:

```
atdl4j-core - the ATDL parsing core
atdl4j-ui - the UI abstraction layer
atdl4j-swt - just the SWT specific code
atdl4j-swing - just the Swing specific code
atdl4j - the drop-in replacement for the current project.
```
